### DISCUSSION TITLE
Pagination with useQueries

### DISCUSSION THREAD

**Discussion by omniqlo** [TanStack/query#2463] [Q&A] [answered]
What’s the best way to do pagination for [dynamic parallel queries](https://react-query.tanstack.com/guides/parallel-queries#dynamic-parallel-queries-with-usequeries)?

For example, let's say that you have an array of 100 post IDs. You want to fetch the first 10 posts in the array, then when you click on the "Next" button, the next 10 posts are fetched, and so on.

```js
const postIds = [1, 2, ..., 100];
const [currentIndex, setCurrentIndex] = useState(0);

const postsQuery = useQueries(
	postIds.slice(currentIndex, currentIndex + 10).map((postId) => ({
		queryKey: ["post", postId],
		queryFn: () => fetchPostById(postId),
		keepPreviousData: true,
	})),
)

const handleNext = () => {
	setCurrentIndex((idx) => idx + 10);
};
```

With the setup above, I'm able to fetch the first 10 posts and then click on the "Next" button (which runs the `handleNext` function) to fetch the next 10 posts. However, this doesn't seem to show the old data (the first 10 posts) while the new data (the next 10 posts) are being fetched even though the `keepPreviousData` field is set to true.

How do I achieve fetching the next batch of posts while showing the previous data?

**EDIT**: Created a CodeSandbox of this use case: https://codesandbox.io/s/epic-shape-cgvjt?file=/src/App.tsx

**Comment by TkDodo** [comment, marked as answer]
`keepPreviousData` doesn't yet work with useQueries. There is an [issue](https://github.com/tannerlinsley/react-query/issues/2181) and an [open PR](https://github.com/tannerlinsley/react-query/pull/2340) to fix it

**Reply by omniqlo** [reply]
Ok, thank you for the details.
